### ISSUE TITLE
Could not pass event PlayerTeleportEvent/PlayerQuitEvent to PerWorldInventory

### ISSUE THREAD

**Issue by mibby** [Gnat008/PerWorldInventory#262]
PWI v1.10.0
```
[05:22:25] [Server thread/ERROR]: Could not pass event PlayerTeleportEvent to PerWorldInventory v1.10.0
org.bukkit.event.EventException: null
	at org.bukkit.plugin.EventExecutor$2.execute(EventExecutor.java:72) ~[patched_1.12.2.jar:git-Paper-1220]
	at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:74) ~[patched_1.12.2.jar:git-Paper-1220]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[patched_1.12.2.jar:git-Paper-1220]
	at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:514) [patched_1.12.2.jar:git-Paper-1220]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:499) [patched_1.12.2.jar:git-Paper-1220]
	at org.bukkit.craftbukkit.v1_12_R1.entity.CraftPlayer.teleport(CraftPlayer.java:608) [patched_1.12.2.jar:git-Paper-1220]
	at com.earth2me.essentials.spawn.EssentialsSpawnPlayerListener$2.run(EssentialsSpawnPlayerListener.java:86) [EssentialsXSpawn.jar:?]
	at org.bukkit.craftbukkit.v1_12_R1.scheduler.CraftTask.run(CraftTask.java:58) [patched_1.12.2.jar:git-Paper-1220]
	at org.bukkit.craftbukkit.v1_12_R1.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:356) [patched_1.12.2.jar:git-Paper-1220]
	at net.minecraft.server.v1_12_R1.MinecraftServer.D(MinecraftServer.java:833) [patched_1.12.2.jar:git-Paper-1220]
	at net.minecraft.server.v1_12_R1.DedicatedServer.D(DedicatedServer.java:426) [patched_1.12.2.jar:git-Paper-1220]
	at net.minecraft.server.v1_12_R1.MinecraftServer.C(MinecraftServer.java:767) [patched_1.12.2.jar:git-Paper-1220]
	at net.minecraft.server.v1_12_R1.MinecraftServer.run(MinecraftServer.java:665) [patched_1.12.2.jar:git-Paper-1220]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_144]
Caused by: java.lang.NullPointerException
	at me.gnat008.perworldinventory.data.players.PWIPlayerManager.updateCache(PWIPlayerManager.java:404) ~[?:?]
	at me.gnat008.perworldinventory.data.players.PWIPlayerManager.addPlayer(PWIPlayerManager.java:116) ~[?:?]
	at me.gnat008.perworldinventory.listeners.player.PlayerTeleportListener.onPlayerTeleport(PlayerTeleportListener.java:51) ~[?:?]
	at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor167.execute(Unknown Source) ~[?:?]
	at org.bukkit.plugin.EventExecutor$2.execute(EventExecutor.java:70) ~[patched_1.12.2.jar:git-Paper-1220]
	... 13 more
```
```
[05:02:59] [Server thread/ERROR]: Could not pass event PlayerTeleportEvent to PerWorldInventory v1.10.0
org.bukkit.event.EventException: null
	at org.bukkit.plugin.EventExecutor$2.execute(EventExecutor.java:72) ~[patched_1.12.2.jar:git-Paper-1220]
	at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:74) ~[patched_1.12.2.jar:git-Paper-1220]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[patched_1.12.2.jar:git-Paper-1220]
	at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:514) [patched_1.12.2.jar:git-Paper-1220]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:499) [patched_1.12.2.jar:git-Paper-1220]
	at org.bukkit.craftbukkit.v1_12_R1.entity.CraftPlayer.teleport(CraftPlayer.java:608) [patched_1.12.2.jar:git-Paper-1220]
	at org.bukkit.craftbukkit.v1_12_R1.entity.CraftEntity.teleport(CraftEntity.java:303) [patched_1.12.2.jar:git-Paper-1220]
	at com.onarandombox.MultiverseCore.utils.SimpleSafeTTeleporter.safelyTeleport(SimpleSafeTTeleporter.java:215) [Multiverse-Core-2.6.0-SNAPSHOT.jar:?]
	at com.onarandombox.MultiverseCore.utils.SimpleSafeTTeleporter.teleport(SimpleSafeTTeleporter.java:335) [Multiverse-Core-2.6.0-SNAPSHOT.jar:?]
	at com.onarandombox.MultiverseCore.commands.TeleportCommand.runCommand(TeleportCommand.java:163) [Multiverse-Core-2.6.0-SNAPSHOT.jar:?]
	at com.pneumaticraft.commandhandler.multiverse.CommandHandler.checkAndRunCommand(CommandHandler.java:279) [Multiverse-Core-2.6.0-SNAPSHOT.jar:?]
	at com.pneumaticraft.commandhandler.multiverse.CommandHandler.processFoundCommands(CommandHandler.java:143) [Multiverse-Core-2.6.0-SNAPSHOT.jar:?]
	at com.pneumaticraft.commandhandler.multiverse.CommandHandler.locateAndRunCommand(CommandHandler.java:93) [Multiverse-Core-2.6.0-SNAPSHOT.jar:?]
	at com.onarandombox.MultiverseCore.MultiverseCore.onCommand(MultiverseCore.java:919) [Multiverse-Core-2.6.0-SNAPSHOT.jar:?]
	at org.bukkit.command.PluginCommand.execute(PluginCommand.java:44) [patched_1.12.2.jar:git-Paper-1220]
	at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:151) [patched_1.12.2.jar:git-Paper-1220]
	at org.bukkit.craftbukkit.v1_12_R1.CraftServer.dispatchCommand(CraftServer.java:676) [patched_1.12.2.jar:git-Paper-1220]
	at net.minecraft.server.v1_12_R1.PlayerConnection.handleCommand(PlayerConnection.java:1431) [patched_1.12.2.jar:git-Paper-1220]
	at net.minecraft.server.v1_12_R1.PlayerConnection.chat(PlayerConnection.java:1340) [patched_1.12.2.jar:git-Paper-1220]
	at org.bukkit.craftbukkit.v1_12_R1.entity.CraftPlayer.chat(CraftPlayer.java:337) [patched_1.12.2.jar:git-Paper-1220]
	at com.gmail.filoghost.chestcommands.internal.icon.command.PlayerIconCommand.execute(PlayerIconCommand.java:15) [ChestCommands.jar:?]
	at com.gmail.filoghost.chestcommands.internal.CommandsClickHandler.onClick(CommandsClickHandler.java:34) [ChestCommands.jar:?]
	at com.gmail.filoghost.chestcommands.api.Icon.onClick(Icon.java:307) [ChestCommands.jar:?]
	at com.gmail.filoghost.chestcommands.internal.icon.ExtendedIcon.onClick(ExtendedIcon.java:224) [ChestCommands.jar:?]
	at com.gmail.filoghost.chestcommands.task.ExecuteCommandsTask.run(ExecuteCommandsTask.java:21) [ChestCommands.jar:?]
	at org.bukkit.craftbukkit.v1_12_R1.scheduler.CraftTask.run(CraftTask.java:58) [patched_1.12.2.jar:git-Paper-1220]
	at org.bukkit.craftbukkit.v1_12_R1.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:356) [patched_1.12.2.jar:git-Paper-1220]
	at net.minecraft.server.v1_12_R1.MinecraftServer.D(MinecraftServer.java:833) [patched_1.12.2.jar:git-Paper-1220]
	at net.minecraft.server.v1_12_R1.DedicatedServer.D(DedicatedServer.java:426) [patched_1.12.2.jar:git-Paper-1220]
	at net.minecraft.server.v1_12_R1.MinecraftServer.C(MinecraftServer.java:767) [patched_1.12.2.jar:git-Paper-1220]
	at net.minecraft.server.v1_12_R1.MinecraftServer.run(MinecraftServer.java:665) [patched_1.12.2.jar:git-Paper-1220]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_144]
Caused by: java.lang.NullPointerException
	at me.gnat008.perworldinventory.data.players.PWIPlayerManager.updateCache(PWIPlayerManager.java:404) ~[?:?]
	at me.gnat008.perworldinventory.data.players.PWIPlayerManager.addPlayer(PWIPlayerManager.java:116) ~[?:?]
	at me.gnat008.perworldinventory.listeners.player.PlayerTeleportListener.onPlayerTeleport(PlayerTeleportListener.java:51) ~[?:?]
	at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor167.execute(Unknown Source) ~[?:?]
	at org.bukkit.plugin.EventExecutor$2.execute(EventExecutor.java:70) ~[patched_1.12.2.jar:git-Paper-1220]
	... 31 more

```

**Comment by mibby** [comment]
```
[05:46:44] [Server thread/ERROR]: Could not pass event PlayerQuitEvent to PerWorldInventory v1.10.0
org.bukkit.event.EventException: null
	at org.bukkit.plugin.EventExecutor$2.execute(EventExecutor.java:72) ~[patched_1.12.2.jar:git-Paper-1220]
	at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:74) ~[patched_1.12.2.jar:git-Paper-1220]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[patched_1.12.2.jar:git-Paper-1220]
	at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:514) [patched_1.12.2.jar:git-Paper-1220]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:499) [patched_1.12.2.jar:git-Paper-1220]
	at net.minecraft.server.v1_12_R1.PlayerList.disconnect(PlayerList.java:423) [patched_1.12.2.jar:git-Paper-1220]
	at net.minecraft.server.v1_12_R1.PlayerConnection.a(PlayerConnection.java:1111) [patched_1.12.2.jar:git-Paper-1220]
	at net.minecraft.server.v1_12_R1.NetworkManager.handleDisconnection(NetworkManager.java:290) [patched_1.12.2.jar:git-Paper-1220]
	at net.minecraft.server.v1_12_R1.ServerConnection.c(ServerConnection.java:184) [patched_1.12.2.jar:git-Paper-1220]
	at net.minecraft.server.v1_12_R1.MinecraftServer.D(MinecraftServer.java:940) [patched_1.12.2.jar:git-Paper-1220]
	at net.minecraft.server.v1_12_R1.DedicatedServer.D(DedicatedServer.java:426) [patched_1.12.2.jar:git-Paper-1220]
	at net.minecraft.server.v1_12_R1.MinecraftServer.C(MinecraftServer.java:767) [patched_1.12.2.jar:git-Paper-1220]
	at net.minecraft.server.v1_12_R1.MinecraftServer.run(MinecraftServer.java:665) [patched_1.12.2.jar:git-Paper-1220]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_144]
Caused by: java.lang.NullPointerException
	at me.gnat008.perworldinventory.data.players.PWIPlayerManager.updateCache(PWIPlayerManager.java:404) ~[?:?]
	at me.gnat008.perworldinventory.process.PlayerQuitProcess.processPlayerLeave(PlayerQuitProcess.java:41) ~[?:?]
	at me.gnat008.perworldinventory.listeners.player.PlayerQuitListener.onPlayerQuit(PlayerQuitListener.java:42) ~[?:?]
	at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor170.execute(Unknown Source) ~[?:?]
	at org.bukkit.plugin.EventExecutor$2.execute(EventExecutor.java:70) ~[patched_1.12.2.jar:git-Paper-1220]
	... 13 more
```

**Comment by Gnat008** [comment]
Do you have the economy functions enabled?

**Comment by mibby** [comment]
I do not, no.

```
player:
    # Save and load players' economy balances. Requires Vault!
    economy: false
```

**Comment by Gnat008** [comment]
I'd guess that this is caused by not having an Economy plugin available. But there is already a null check there... Unless Vault is installed, and is hooking into it successfully. Hm.

**Comment by mibby** [comment]
I do have vault installed with Essentials, but I don't use economy functionality in the traditional sense. I recently installed TokenManager and enabled their vault-hook, having all economy rewarding for events and arenas provide token points through vault instead of soft currency.

https://www.spigotmc.org/resources/tokenmanager.8610/
```# [----------------------------] !!! WARNING !!! [----------------------------]
# 1) All of your previous economy data will be converted to their tokenmanager economy data,
# and commands such as '/eco (essentials)' will no longer function when player is offline.
# please use '/tokenmanager add | remove | set' instead, which supports offline players.
#
# 2) When you have vault hook enabled + have attempted to reset the economy,
# essentials economy data will load as tokenmanager data, which means they
# will still load balances from their essentials userdata file. Best to reset
# essentials economy before resetting tokenmanager data.
#
#  Please be careful with this option!
#    YOU HAVE BEEN WARNED...
# [----------------------------] !!! WARNING !!! [----------------------------]
# Read the comments above if you haven't already.
# If enabled, TokenManager's economy will be used by all of your
# plugins with Vault support, including Essentials.
vault-hook: true
```

**Comment by mibby** [comment]
@Gnat008 Still happening with v1.12-beta.1

```
[10:47:22] [Server thread/ERROR]: Could not pass event PlayerQuitEvent to PerWorldInventory v1.12-beta.1
org.bukkit.event.EventException: null
	at org.bukkit.plugin.EventExecutor$2.execute(EventExecutor.java:72) ~[patched_1.12.2.jar:git-Paper-1234]
	at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:74) ~[patched_1.12.2.jar:git-Paper-1234]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[patched_1.12.2.jar:git-Paper-1234]
	at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:514) [patched_1.12.2.jar:git-Paper-1234]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:499) [patched_1.12.2.jar:git-Paper-1234]
	at net.minecraft.server.v1_12_R1.PlayerList.disconnect(PlayerList.java:423) [patched_1.12.2.jar:git-Paper-1234]
	at net.minecraft.server.v1_12_R1.PlayerConnection.a(PlayerConnection.java:1139) [patched_1.12.2.jar:git-Paper-1234]
	at net.minecraft.server.v1_12_R1.NetworkManager.handleDisconnection(NetworkManager.java:338) [patched_1.12.2.jar:git-Paper-1234]
	at net.minecraft.server.v1_12_R1.ServerConnection.c(ServerConnection.java:184) [patched_1.12.2.jar:git-Paper-1234]
	at net.minecraft.server.v1_12_R1.MinecraftServer.D(MinecraftServer.java:940) [patched_1.12.2.jar:git-Paper-1234]
	at net.minecraft.server.v1_12_R1.DedicatedServer.D(DedicatedServer.java:426) [patched_1.12.2.jar:git-Paper-1234]
	at net.minecraft.server.v1_12_R1.MinecraftServer.C(MinecraftServer.java:767) [patched_1.12.2.jar:git-Paper-1234]
	at net.minecraft.server.v1_12_R1.MinecraftServer.run(MinecraftServer.java:665) [patched_1.12.2.jar:git-Paper-1234]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_144]
Caused by: java.lang.NullPointerException
	at me.gnat008.perworldinventory.data.players.PWIPlayerManager.updateCache(PWIPlayerManager.java:430) ~[?:?]
	at me.gnat008.perworldinventory.process.PlayerQuitProcess.processPlayerLeave(PlayerQuitProcess.java:41) ~[?:?]
	at me.gnat008.perworldinventory.listeners.player.PlayerQuitListener.onPlayerQuit(PlayerQuitListener.java:42) ~[?:?]
	at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor171.execute(Unknown Source) ~[?:?]
	at org.bukkit.plugin.EventExecutor$2.execute(EventExecutor.java:70) ~[patched_1.12.2.jar:git-Paper-1234]
	... 13 more
```

**Comment by Gnat008** [comment]
@mibby Could you try this beta version? Let's see if it was just the bank causing the issue. https://github.com/Gnat008/PerWorldInventory/releases/tag/v1.12-beta.2

**Comment by mibby** [comment]
@Gnat008 I have not seen the error again since updating so I assume that was it. Thanks!

**Comment by Gnat008** [comment]
Fixed by 6c77f85
